test/notifications-service-test.js

```diff
@@ -1,171 +1,131 @@
+	browser.tabs.create = sinon.stub().resolves(true);
 
-	window.chrome.permissions.contains = sinon.stub().yieldsAsync(true);
-	window.chrome.tabs.query = sinon.stub().yieldsAsync([]);
-	window.fetch = sinon.stub().returns(Promise.resolve(t.context.defaultResponse));
-	window.localStorage.getItem.withArgs(t.context.notificationId).returns(t.context.notificationsUrl);
+	browser.notifications.create = sinon.stub().resolves(t.context.notificationId);
+	browser.notifications.clear = sinon.stub().resolves(true);
 
-	await service.openNotification(t.context.notificationId);
+	browser.permissions.contains = sinon.stub().yieldsAsync(true);
 
-	t.true(window.localStorage.getItem.calledWith(t.context.notificationId));
+	browser.storage.sync.set({
+		options: {
+			token: 'a1b2c3d4e5f6g7h8i9j0a1b2c3d4e5f6g7h8i9j0',
+			rootUrl: 'https://api.github.com/',
+			playNotifSound: true,
+			showDesktopNotif: true
+		}
+	});
 });
 
-test('#openNotification clears notification from queue by notificationId', async t => {
-	const service = t.context.service;
-
-	window.chrome.permissions.contains = sinon.stub().yieldsAsync(true);
-	window.chrome.tabs.query = sinon.stub().yieldsAsync([]);
-	window.fetch = sinon.stub().returns(Promise.resolve(t.context.defaultResponse));
-	window.localStorage.getItem.withArgs(t.context.notificationId).returns(t.context.notificationsUrl);
+test.serial('#openNotification gets notification url by notificationId from local-store', async t => {
+	const {service, notificationId} = t.context;
 
-	await service.openNotification(t.context.notificationId);
+	await service.openNotification(notificationId);
 
-	t.true(window.chrome.notifications.clear.calledWithMatch(t.context.notificationId));
+	t.true(browser.storage.local.get.calledWith(notificationId));
 });
 
-test.serial('#openNotification skips network requests if no url returned by PersistenceService', async t => {
-	const service = t.context.service;
+test.serial('#openNotification clears notification from queue by notificationId', async t => {
+	const {service, notificationId} = t.context;
 
-	window.localStorage.getItem = sinon.stub().returns(null);
-	window.fetch = sinon.stub().returns(Promise.resolve(t.context.defaultResponse));
+	await service.openNotification(notificationId);
 
-	await service.openNotification(t.context.notificationId);
-
-	t.is(window.fetch.callCount, 0);
+	t.true(browser.notifications.clear.calledWithMatch(notificationId));
 });
 
-test('#openNotification closes notification if no url returned by PersistenceService', async t => {
-	const service = t.context.service;
-
-	window.localStorage.getItem = sinon.stub().returns(null);
-	service.closeNotification = sinon.stub().returns(Promise.resolve());
+test.serial('#openNotification skips network requests if no url returned by local-store', async t => {
+	const {service} = t.context;
 
-	await service.openNotification(t.context.notificationId);
+	await service.openNotification('random-notification-id');
```

Comment:
That string is not random at all now, some of the test may pass by accident because of this.

Reply:
Err, that should be 'non-existent-notification-id'.that is not added to local store by `browser.storage.local.set`.